Regarding the broken tests:

What follows is a copy from the CI. 

Functionality from @denizural:
```
FAILED tests/test_esm_runscripts/test_filedicts.py::test_ln_raises_exception_when_pointing_to_itself
[343](https://github.com/esm-tools/esm_tools/runs/7878295110?check_suite_focus=true#step:7:344)
FAILED tests/test_esm_runscripts/test_filedicts.py::test_ln_raises_exception_when_target_is_a_directory_and_not_a_file
[344](https://github.com/esm-tools/esm_tools/runs/7878295110?check_suite_focus=true#step:7:345)
FAILED tests/test_esm_runscripts/test_filedicts.py::test_ln_raises_exception_when_target_path_exists
```
This relates to the fact that I had to move the stamping above your checks. If you have the same style for the `ln` function as we do for `cp` and `mv` that might work.

These ones from @mandresm I need to look at in detail:
```
FAILED tests/test_esm_runscripts/test_filedicts.py::test_check_file_syntax_type_missing
[345](https://github.com/esm-tools/esm_tools/runs/7878294985?check_suite_focus=true#step:7:346)
FAILED tests/test_esm_runscripts/test_filedicts.py::test_check_file_syntax_type_incorrect
[346](https://github.com/esm-tools/esm_tools/runs/7878294985?check_suite_focus=true#step:7:347)
FAILED tests/test_esm_runscripts/test_filedicts.py::test_check_file_syntax_input
[347](https://github.com/esm-tools/esm_tools/runs/7878294985?check_suite_focus=true#step:7:348)
FAILED tests/test_esm_runscripts/test_filedicts.py::test_check_file_syntax_output
```

--- a/src/esm_runscripts/filedicts.py
+++ b/src/esm_runscripts/filedicts.py
@@ -172,7 +172,7 @@ class SimulationFile(dict):
         return super().__setattr__(name, value)
 
     def __setitem__(self, key: Any, value: Any) -> None:
-        # Checks for changing with sim_file["my_key"] = "new_value"
+        """Checks for changing with sim_file["my_key"] = "new_value""""
         if key == "datestamp_format":
             self._check_datestamp_format_is_allowed(value)
         if key == "datestamp_method":

--- a/src/esm_runscripts/filedicts.py
+++ b/src/esm_runscripts/filedicts.py
@@ -164,7 +164,7 @@ class SimulationFile(dict):
         self._check_path_in_computer_is_abs()
 
     def __setattr__(self, name: str, value: Any) -> None:
-        # Checks when changing dot attributes for disallowed values:
+        """Checks when changing dot attributes for disallowed values"""
         if name == "datestamp_format":
             self._check_datestamp_format_is_allowed(value)
         if name == "datestamp_method":

--- a/tests/test_esm_runscripts/test_filedicts.py
+++ b/tests/test_esm_runscripts/test_filedicts.py
@@ -441,6 +441,7 @@ def test_ln_raises_exception_when_target_path_does_not_exist(simulation_file, fs
 
 # ========== end of ln() tests ==========
 
+
 def test_check_file_syntax_type_missing():
     """Tests for ``type`` variable missing"""
     dummy_config = """
@@ -455,7 +456,9 @@ def test_check_file_syntax_type_missing():
         experiment_input_dir: /work/ollie/pgierz/some_exp/input/echam
         thisrun_input_dir: /work/ollie/pgierz/some_exp/run_20010101-20010101/input/echam
     """
+    date = esm_calendar.Date("2000-01-01T00:00:00")
     config = yaml.safe_load(dummy_config)
+    config["general"]["current_date"] = date
 
     # Captures output (i.e. the user-friendly error)
     with Capturing() as output:
@@ -466,6 +469,7 @@ def test_check_file_syntax_type_missing():
 
     error_text = "the \x1b[31mtype\x1b[0m variable is missing"
     assert any([error_text in line for line in output])
+
 
 def test_check_file_syntax_type_incorrect():
     """Tests for ``type`` variable being incorrectly defined"""
@@ -481,7 +485,9 @@ def test_check_file_syntax_type_incorrect():
         experiment_input_dir: /work/ollie/pgierz/some_exp/input/echam
         thisrun_input_dir: /work/ollie/pgierz/some_exp/run_20010101-20010101/input/echam
     """
+    date = esm_calendar.Date("2000-01-01T00:00:00")
     config = yaml.safe_load(dummy_config)
+    config["general"]["current_date"] = date
 
     # Captures output (i.e. the user-friendly error)
     with Capturing() as output:
@@ -492,6 +498,7 @@ def test_check_file_syntax_type_incorrect():
 
     error_text = "is_wrong\x1b[0m is not a supported \x1b[31mtype"
     assert any([error_text in line for line in output])
+
 
 def test_check_file_syntax_input():
     """
@@ -509,7 +516,9 @@ def test_check_file_syntax_input():
         experiment_input_dir: /work/ollie/pgierz/some_exp/input/echam
         thisrun_input_dir: /work/ollie/pgierz/some_exp/run_20010101-20010101/input/echam
     """
+    date = esm_calendar.Date("2000-01-01T00:00:00")
     config = yaml.safe_load(dummy_config)
+    config["general"]["current_date"] = date
 
     # Captures output (i.e. the user-friendly error)
     with Capturing() as output:
@@ -522,6 +531,7 @@ def test_check_file_syntax_input():
     assert any([error_text in line for line in output])
     error_text = "the \x1b[31mname_in_computer\x1b[0m variable is missing"
     assert any([error_text in line for line in output])
+
 
 def test_check_file_syntax_output():
     """Tests for missing ``name_in_work`` for output file types"""
@@ -537,7 +547,9 @@ def test_check_file_syntax_output():
         experiment_input_dir: /work/ollie/pgierz/some_exp/input/echam
         thisrun_input_dir: /work/ollie/pgierz/some_exp/run_20010101-20010101/input/echam
     """
+    date = esm_calendar.Date("2000-01-01T00:00:00")
     config = yaml.safe_load(dummy_config)
+    config["general"]["current_date"] = date
 
     # Captures output (i.e. the user-friendly error)
     with Capturing() as output:
@@ -548,6 +560,7 @@ def test_check_file_syntax_output():
 
     error_text = "the \x1b[31mname_in_work\x1b[0m variable is missing"
     assert any([error_text in line for line in output])
+
 
 def test_check_path_in_computer_is_abs(fs):
     """

